packages/runtime/src/common/url.ts

```diff
@@ -4,11 +4,17 @@ export const isBpUrl = (str: string): boolean => {
   return re.test(str)
 }
 
-// Duplicate of modules/builtin/src/content-types/_utils.js
 export const formatUrl = (baseUrl: string, url: string): string => {
-  if (isBpUrl(url)) {
-    return `${baseUrl}${url}`
-  } else {
+  const bpUrlRegex = /^\/api\/.*\/bots\/(.*)\/media\/(.*)/g
+  const parts = bpUrlRegex.exec(url)
```

Comment:
Can we extract this in a function to make it more readable and more useable. What do you think of to following

```ts
interface MediaUrlElements {
  botId: string
  mediaId: string
  isBotpressUrl: boolean
}

function parseBotpressMediaUrl(url:string ): MediaUrlElements {
  let botId, mediaId = null

  const bpUrlRegex = /^\/api\/.*\/bots\/(.*)\/media\/(.*)/g
  const parts = bpUrlRegex.exec(url)
  if (parts) {
    botId = parts[1]
    mediaId = parts[2]
  }

  return { botId, mediaId, isBotpressURL: !!parts }
}
```

Much more explicit and easy to use in the format urkl function

Reply:
we could then get rid of the isBPUrl function and use only this one :), test it and make it bullet proof